### ISSUE TITLE
CreateSnapshot using govcsim

### ISSUE THREAD

**Issue by Akasurde** [vmware/govmomi#796]
DC0_H0_VM0 is emulated by govcsim.

I am unable to create snapshot using govcsim emulated VM, when using 

```
$ go run main.go snapshot.create -vm=DC0_H0_VM0 SnapShot_0001
```

I get following warning -

```
VirtualMachine:virtualmachine-61 does not implement: CreateSnapshot_Task
```

**Comment by dougm** [comment]
This would be a start, but you probably want to populate the `vm.Snapshot` property too.

```go
func (vm *VirtualMachine) CreateSnapshotTask(req *types.CreateSnapshot_Task) soap.HasFault {
	task := CreateTask(vm, "createSnapshot", func(t *Task) (types.AnyType, types.BaseMethodFault) {

		return nil, nil
	})

	task.Run()

	return &methods.CreateSnapshot_TaskBody{
		Res: &types.CreateSnapshot_TaskResponse{
			Returnval: task.Self,
		},
	}
}
```

**Comment by Akasurde** [comment]
@dougm Thanks for snippet. Could you please elaborate more `vm.Snapshot` ? I am not that well versed with code. I am wondering - how to keep record of snapshot tree in memory?

**Comment by dougm** [comment]
In general, you can modify the `vm.VirtualMachine.Snapshot` field directly within the `CreateTask` function above.

I would start by seing how it looks against a real ESX or VC:

```console
% vm=foo
% govc vm.create -disk 1G $vm                                                                                                                                                                                                                                     
% govc vm.info -json $vm | jq .VirtualMachines[].Snapshot                                                                                                                                                                                                          
null
% govc  snapshot.create -vm $vm 1st
% govc snapshot.tree -vm $vm
1st
.
% govc vm.info -json $vm | jq .VirtualMachines[].Snapshot
{
  "CurrentSnapshot": {
    "Type": "VirtualMachineSnapshot",
    "Value": "285-snapshot-1"
  },
  "RootSnapshotList": [
    {
      "Snapshot": {
        "Type": "VirtualMachineSnapshot",
        "Value": "285-snapshot-1"
      },
      "Vm": {
        "Type": "VirtualMachine",
        "Value": "285"
      },
      "Name": "1st",
      "Description": "",
      "Id": 1,
      "CreateTime": "2017-07-26T17:50:59.230273Z",
      "State": "poweredOn",
      "Quiesced": false,
      "BackupManifest": "",
      "ChildSnapshotList": null,
      "ReplaySupported": false
    }
  ]
}
```

Then look at details on the snapshot(s) themselves if needed:
```console
% govc object.collect VirtualMachineSnapshot:285-snapshot-1 
availableField  []types.CustomFieldDef          
childSnapshot   []types.ManagedObjectReference  
config          types.VirtualMachineConfigInfo  ...
value           []types.BaseCustomFieldValue    
vm              types.ManagedObjectReference    VirtualMachine:285
```

**Comment by dougm** [comment]
Fixed in #855
